src/com/amazon/corretto/crypto/provider/AesCbcSpi.java

```diff
@@ -142,14 +143,29 @@ private boolean noPadding() {
 
   @Override
   protected void engineSetMode(final String mode) throws NoSuchAlgorithmException {
-    // no op. One only needs to provide an implementation if the same Spi class instance can be used
-    // for different modes.
+    if (!"CBC".equalsIgnoreCase(mode)) {
+      throw new NoSuchAlgorithmException("Only CBC mode is supported.");
+    }
   }
 
   @Override
   protected void engineSetPadding(final String padding) throws NoSuchPaddingException {
```

Comment:
There are two variables named `padding`. `private int padding;` on line 100, and `final String padding` here on line 152. Can you rename one of them so that they don't have the same name?

Reply:
Took this a step further and replaced the field's type with the Padding enum to give us a bit more compile time guarantees around the possible padding values passed to AWS-LC.